wgpu/src/lib.rs

```diff
@@ -2124,6 +2124,31 @@ impl Device {
         }
     }
 
+    /// Creates a shader module from either SPIR-V or WGSL source code without uniformity checks.
+    ///
+    /// # Safety
+    /// In contrast with [`create_shader_module`](Self::create_shader_module) this function
+    /// creates a shader module without unifromity checks which allows shaders to perform
+    /// operations which can lead to undefined behavior like data races, thus it's
+    /// the caller responsibility to pass a shader which doesn't perform any of these
+    /// operations.
+    ///
+    /// This is equivalent to `create_shader_module` effect on web.
```

Comment:
This isn't actually equivalent. With disabling bounds checks, it doesn't actually change the validity of the shader. This method makes otherwise invalid shaders valid. I think we need a proper feature for this. Maybe `UNCHECKED_SHADERS` to cover any future validation restrictions too.

Reply:
Ah, I was coming at this from 'web=webgpu', which should have uniformity analysis forgetting the 'web=webgl' case. Thanks for the reminder

Reply:
My point is more that, when calling *_unchecked, it doesn't matter if there are bounds checks in the shader or not, so the user doesn't need to know if eliding bounds checks are supported.

When calling `*_non_uniform`, it _does_ matter if eliding uniformity analysis is supported, because some shaders won't compile. Because of this, we need some way to inform the user if eliding uniformity analysis is supported on the device.

Reply:
Possibly? I'd hope use of this API would be relatively rare. 

More precisely, I think that the non-uniform part of this API is kind of inherently tied to working around naga, with knowledge of the code it emits to the backend. It's unclear to me what cases are valid wgsl if they don't pass uniformity analysis. That is, if you have a version of your code which uses the fact that uniformity analysis can be disabled, and another one which doesn't, you should always use the one which doesn't.

Reply:
If the workgroupBroadcast proposal is accepted (and implemented in naga), then that would be my preferred way to avoid triggering uniformity analysis errors or warnings, and then I would not use this API. However, it might still make sense as a short term solution.